projects/fal/tests/test_stability.py

```diff
@@ -58,6 +54,7 @@ def calling_function(result):
     assert calling_function(regular_function()) == 84
 
 
+@pytest.mark.xfail(reason="See https://github.com/fal-ai/fal/issues/169")
```

Comment:
This test is using an dill-based fal version and results in an error in nested function that looks something like so:

```
            return_value = _UNSET
            # Allow isolate provided arguments (such as setup function) to take
            # precedence over the ones provided by the user.
            partial_func = _prepare_partial_func(func, *args, **kwargs)
            for partial_result in self._connection.run(
                partial_func,
                environments,
                machine_requirements=machine_requirements,
                setup_function=setup_function,
            ):
                for log in partial_result.logs:
                    self._log_printer.print(log)

                if partial_result.status.state is not HostedRunState.IN_PROGRESS:
                    state = partial_result.status.state
                    if state is HostedRunState.INTERNAL_FAILURE:
                        raise InternalFalServerlessError(
                            "An internal failure occurred while performing this run."
                        )
                    elif state is HostedRunState.SUCCESS:
                        return_value = partial_result.result
                    else:
>                       raise NotImplementedError("Unknown state: ", state)
E                       NotImplementedError: ('Unknown state: ', <HostedRunState.IN_PROGRESS: 0>)

projects/fal/src/fal/api.py:484: NotImplementedError
```

Will likely need a deeper investigation later.